hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactionCommitSink.java

```diff
@@ -132,6 +127,15 @@ private void commitIfNecessary(String instant, Collection<CompactionCommitEvent>
     if (!isReady) {
       return;
     }
+
+    if (events.stream().anyMatch(CompactionCommitEvent::isFailed)) {
+      // handle failure case
+      CompactionUtil.rollbackCompaction(table, instant);
+      // remove commitBuffer avoid commit with preview fileId
+      reset(instant);
+      return;
+    }
+
```

Comment:
Can we ensure the buffer was cleared like this:
```java
try {
  // handle failure case
  CompactionUtil.rollbackCompaction(table, instant);
} finally {
  // remove commitBuffer to avoid obsolete metadata commit
  reset(instant);
  return;
}
```